modules/core/test/test_stl_forward.cpp

```diff
@@ -0,0 +1,217 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+#include "test_precomp.hpp"
+#include "opencv2/core/stl/algorithm.hpp"
+#include "opencv2/core.hpp"
+#include <numeric>
+
+namespace opencv_test { namespace {
+
+// The fixture for testing class Foo.
+class CORE_stl_forward : public ::testing::Test {
+ protected:
+
+    void SetUp() override
+    {
+        size_t size = 13;
+        mat = cv::Mat(int(size),int(size),CV_32S);
+        mat_f = cv::Mat_<float>(int(size),int(size));
+
+
+        intVec = std::vector<int>(size*size);
+        intList = std::list<int>(size*size);
+
+        subMat = mat(cv::Rect(2,2,7,7));
+        matSub_f = mat_f(cv::Rect(2,2,7,7));
+
+        std::iota(mat.begin<int>(), mat.end<int>(), -1);
+        std::iota(mat_f.begin(), mat_f.end(), -1.0f);
+
+        std::iota(intVec.begin(), intVec.end(), -1);
+        std::iota(intList.begin(), intList.end(), -1);
+
+        std::iota(subMat.begin(), subMat.end(), -1);
+        std::iota(matSub_f.begin(), matSub_f.end(), -1.0f);
+    }
+
+    void TearDown() override
+    {
+
+    }
+    cv::Mat mat;
+    cv::Mat_<float> mat_f;
+
+    std::vector<int> intVec;
+    std::list<int> intList;
+
+    cv::Mat_<int> subMat;
+    cv::Mat_<float> matSub_f;
+};
+
+TEST_F(CORE_stl_forward, iterators_replacable)
+{
+        ///////////////////////////////////Contiguous Tests/////////////////////////////////
+    //Test only one "span" where we can replace iterators. Lambda doesn't have an effect
+    EXPECT_TRUE(cv::experimental::__iterators__replaceable(mat.begin<int>(), mat.end<int>(),[](int val){return val*2;}));
+
+    //Should be false because there is no cv::Mat iterator. Lambda doesn't have an effect
+    EXPECT_FALSE(cv::experimental::__iterators__replaceable(intVec.begin(), intVec.end(),[](int val){return val*2;}));
+
+    //Should be true because there is only one cv::Mat iterator of a contiguous iterator. Lambda doesn't have an effect
+    EXPECT_TRUE(cv::experimental::__iterators__replaceable(intVec.begin(), intVec.end(), mat.begin<int>(),[](int val){return val*2;}));
+
+    ///////////////////////////////////Submatrix Tests/////////////////////////////////
+    //Test only one "span" where we can't replace iterators (not contiguous)
+    EXPECT_FALSE(cv::experimental::__iterators__replaceable(subMat.begin(), subMat.end()));
+
+    //Should be false because there is a cv::Mat iterator of a non-contiguous matrix
+    EXPECT_FALSE(cv::experimental::__iterators__replaceable(intVec.begin(), intVec.end(), subMat.begin()));
+
+    //Should be false because there is a cv::Mat iterator of a non-contiguous matrix
+    EXPECT_FALSE(cv::experimental::__iterators__replaceable(intVec.begin(), intVec.end(), subMat.begin()));
+}
+
+TEST_F(CORE_stl_forward, DISABLED_iterators_replacable_reverse_iterator)
```

Comment:
Why is this test disabled?

Reply:
Hello @alalek,
thanks for replying. 
The branch is currently only experimental and there is no support for reverse iterators, yet.
I wanted to test the waters and see, if such a feature would be interesting for the community. If this is the case I am willing to add the missing algorithms (that exist in C++11), add support for the reverse iterators and write some documentation for it.